core/models.py

```diff
@@ -212,6 +217,9 @@ def accept_invitation(self, invitation, *, circles):
                 connection=connection.opposite,
             )
 
+        if not invitation.is_open:
```

Comment:
deletion after success, assuming invitation is not expired and connection has been made.